tests/internal/test_native.py

```diff
@@ -0,0 +1,41 @@
+from ddtrace.internal.native import get_configuration_from_disk
+
+
+def test_get_configuration_from_disk(tmp_path):
+    # First test -- config matches & should be returned
+    config_1 = tmp_path / "config_1.yaml"
+    config_1.write_text(
+        """
+rules:
+  - selectors:
+    - origin: language
+      matches:
+        - python
+      operator: equals
+    configuration:
+      DD_SERVICE: my-service
+""",
+        encoding="utf-8",
+    )
+
+    config = get_configuration_from_disk(file_override=str(config_1))
+    assert config == {"DD_SERVICE": "my-service"}
+
+    # Second test -- config does not match & should not be returned
+    config_2 = tmp_path / "config_2.yaml"
+    config_2.write_text(
+        """
+rules:
+  - selectors:
+    - origin: language
+      matches:
+        - nodejs
+      operator: equals
+    configuration:
+      DD_SERVICE: my-service
```

Comment:
In the python tracer the DD_SERVICE environment variable can be overridden by [integrations](https://github.com/DataDog/dd-trace-py/blob/v2.20.0rc1/ddtrace/contrib/internal/flask/patch.py#L66C9-L66C25) and other ddtrace products (ex: [serverless](https://github.com/DataDog/dd-trace-py/blob/v2.20.0rc1/ddtrace/settings/config.py#L483)). 

We will need to audit all the configurations supported by libdatadog and ensure the value set via APM Stable Configurations take precedence.

Reply:
Agreed. I also think that we need to have a discussion on the order of precedence. For instance in the agent, config values set by the runtime (ie inside the agent code) take precedence over Fleet Automation. One could argue that we should replicate that behaviour here; but imo this is a product decision.